flink-formats/flink-protobuf/src/test/java/org/apache/flink/formats/protobuf/Pb3ToRowTest.java

```diff
@@ -85,6 +88,8 @@ public void testDeserialization() throws Exception {
 
         assertEquals(1, rowData.getInt(0));
         assertEquals(2L, rowData.getLong(1));
+
+        assertEquals(ts.getNanos(), row.getTimestamp(11, 3).getNanoOfMillisecond());
```

Comment:
I'm curious how does Timestamp work because I did not define Timestamp type for flink. I guess Timestamp is a message type in protobuf, how come it will convert to flink Timestamp automatically?

Reply:
The org.apache.flink.table.data.binary.BinaryRowData.getTimestamp() method from the flink-table module does the translation to org.apache.flink.table.data.TimestampData.

Reply:
@hdulay I think there is still gap between protobuf and flink internal memory layout. I tried to build a concrete Timestamp like `Timestamp ts =Timestamp.newBuilder().setSeconds(xxx).setNanos(yyy).build()` in unit test. And you can easily find the difference.

Reply:
@maosuhan Yes as I stepped through the ProtobufSQLITCaseTest to add a timestamp field, I realized this issue goes beyond just source generation and compilation errors. Is there an example of how to add new data type?  I'm assuming I'll need to somehow map it to the TimestampType logical type. Thanks

Reply:
Just to step in: adding a new datatype requires a FLIP and a discussion. Why can't any of the existing datatypes be reused?

Reply:
@MartijnVisser Hi. I'm suggesting to add support for 3rd party protobuf datatype. Which needs a mapping to either a RowType or a SimpleType in the protobuf codegen components. I'm assuming this is the correct approach. Thanks

Reply:
@hdulay Yes, I agree with your idea. RowType must be doable to reflect a Timestamp pb type. If you want to make it more easy to use, you can treat `google.protobuf.Timestamp` a special type and manually convert Timestamp to TimestampData in code.

Reply:
Thanks. Let me know if it's appropriate to convert this PR to a draft. Thanks for all the guidance.